backend/migrations/versions/8c79fb896983_add_table_for_buckets.py

```diff
@@ -0,0 +1,206 @@
+        session = orm.Session(bind=bind)
+        datasets: [Dataset] = session.query(Dataset).all()
+        if not has_table('dataset_bucket', engine):
+            op.create_table(
+                'dataset_bucket',
+                sa.Column('bucketUri', sa.String(), nullable=False),
+                sa.Column('label', sa.String(), nullable=False),
+                sa.Column('name', sa.String(), nullable=False),
+                sa.Column('owner', sa.String(), nullable=False),
+                sa.Column('created', sa.DateTime(), nullable=True),
+                sa.Column('updated', sa.DateTime(), nullable=True),
+                sa.Column('deleted', sa.DateTime(), nullable=True),
+                sa.Column('description', sa.String(), nullable=True),
+                sa.Column('tags', postgresql.ARRAY(sa.String()), nullable=True),
+                sa.Column('datasetUri', sa.String(), nullable=False),
+                sa.Column('AwsAccountId', sa.String(), nullable=False),
+                sa.Column('S3BucketName', sa.String(), nullable=False),
+                sa.Column('KmsAlias', sa.String(), nullable=False),
+                sa.Column('imported', sa.Boolean(), nullable=True),
+                sa.Column('importedKmsKey', sa.Boolean(), nullable=True),
+                sa.Column('region', sa.String(), nullable=True),
+                sa.Column('partition', sa.String(), nullable=False, default='aws'),
+                sa.ForeignKeyConstraint(columns=['datasetUri'], refcolumns=['dataset.datasetUri']),
+                sa.PrimaryKeyConstraint('bucketUri'),
+            )
+            print('Creating a new dataset_bucket row for each existing dataset...')
+            for dataset in datasets:
+                dataset_bucket = DatasetBucket(
+                    name=dataset.S3BucketName,
+                    datasetUri=dataset.datasetUri,
+                    AwsAccountId=dataset.AwsAccountId,
+                    S3BucketName=dataset.S3BucketName,
+                    region=dataset.region,
+                    label=dataset.label,
+                    description=dataset.label,
+                    tags=dataset.tags,
+                    owner=dataset.owner,
+                    KmsAlias=dataset.KmsAlias,
+                    imported=dataset.imported,
+                    importedKmsKey=dataset.importedKmsKey,
+                )
+                session.add(dataset_bucket)
+            session.flush()  # flush to get the bucketUri
+
+        for dataset in datasets:
+            shared_bucket_object: ShareObjectItem = session.query(ShareObjectItem).filter(
+                and_(
+                    ShareObjectItem.itemType == ShareableType.S3Bucket.value,
+                    ShareObjectItem.itemUri == dataset.datasetUri,
+                )
+            ).first()
+            dataset_bucket: DatasetBucket = session.query(DatasetBucket).filter(
+                DatasetBucket.datasetUri == dataset.datasetUri
+            ).first()
```

Comment:
I think this is particular to your implementation that had another release. If possible I would remove it, but it is only cosmetic. I don't see it affects other deployments in any way

Reply:
Not particularly sure about what code you are talking about. 

I do see we have `if column_exists('dataset', 'dataSharingModel'):` here. I will remove that from migration script as datasharingModel never made it to open source. Please let me know if you were talking about some other change though!

Reply:
Yes, that and also the following block. For a regular customer, no share_bucket_objects are created at this time, so this block is unnecessary. It is not going to introduce any error as the value checked in the if-clause will be `None`. However, taking the point of view of another customer, I would find it strange and difficult to understand.
```
        for dataset in datasets:
            shared_bucket_object: ShareObjectItem = session.query(ShareObjectItem).filter(
                and_(
                    ShareObjectItem.itemType == ShareableType.S3Bucket.value,
                    ShareObjectItem.itemUri == dataset.datasetUri,
                )
            ).first()
            dataset_bucket: DatasetBucket = session.query(DatasetBucket).filter(
                DatasetBucket.datasetUri == dataset.datasetUri
            ).first()
            if shared_bucket_object is not None:
                shared_bucket_object.itemUri = dataset_bucket.bucketUri
                shared_bucket_object.itemName = dataset_bucket.S3BucketName
```